File: java/0387-first-unique-character-in-a-string.java
class Solution {
    public int firstUniqChar(String s) {
        Map<Character, Integer> count = new HashMap<>();
        for (Character ch : s.toCharArray()) {
            count.put(ch, count.getOrDefault(ch, 0) + 1);
        }
        for (int i = 0; i < s.length(); i++) {
            if (count.get(s.charAt(i)) == 1) {
                return i;
            }
        }
        return -1;
    }
}

File: kotlin/0387-first-unique-character-in-a-string.kt
class Solution {
    fun firstUniqChar(s: String): Int {
        var count = IntArray (26)
        for (c in s)
            count[c - 'a']++

        for ((i, c) in s.withIndex()) {
            if (count[c - 'a'] == 1)
                return i
        }
 
        return -1
    }
}
